src/libraries/System.Private.CoreLib/src/System/Diagnostics/CodeAnalysis/ExperimentalAttribute.cs

```diff
@@ -1,54 +1,55 @@
-///  a diagnostic that indicates that an experimental feature is used. Authors can use this attribute to ship preview features in their assemblies.
-/// </summary>
-[AttributeUsage(AttributeTargets.Assembly |
-                AttributeTargets.Module |
-                AttributeTargets.Class |
-                AttributeTargets.Struct |
-                AttributeTargets.Enum |
-                AttributeTargets.Constructor |
-                AttributeTargets.Method |
-                AttributeTargets.Property |
-                AttributeTargets.Field |
-                AttributeTargets.Event |
-                AttributeTargets.Interface |
-                AttributeTargets.Delegate, Inherited = false)]
-public sealed class ExperimentalAttribute : Attribute
+namespace System.Diagnostics.CodeAnalysis
 {
     /// <summary>
-    ///  Initializes a new instance of the <see cref="ExperimentalAttribute"/> class, specifying the ID that the compiler will use
-    ///  when reporting a use of the API the attribute applies to.
-    /// </summary>
-    /// <param name="diagnosticId">The ID that the compiler will use when reporting a use of the API the attribute applies to.</param>
-    /// <exception cref="ArgumentNullException"><paramref name="diagnosticId"/> is <see langword="null"/>.</exception>
-    /// <exception cref="ArgumentException"><paramref name="diagnosticId"/> is empty.</exception>
-    public ExperimentalAttribute(string diagnosticId)
-    {
-        ArgumentException.ThrowIfNullOrEmpty(diagnosticId);
-        DiagnosticId = diagnosticId;
-    }
-
-    /// <summary>
-    ///  Gets or sets the ID that the compiler will use when reporting a use of the API the attribute applies to.
+    ///  Indicates that an API is experimental and it may change in the future.
     /// </summary>
-    /// <value>The unique diagnostic ID.</value>
     /// <remarks>
-    ///  The diagnostic ID is shown in build output for warnings and errors.
-    ///  <para>This property represents the unique ID that can be used to suppress the warnings or errors, if needed.</para>
+    ///   This attribute allows call sites to be flagged with a diagnostic that indicates that an experimental
+    ///   feature is used. Authors can use this attribute to ship preview features in their assemblies.
     /// </remarks>
-    public string DiagnosticId { get; }
+    [AttributeUsage(AttributeTargets.Assembly |
+                    AttributeTargets.Module |
+                    AttributeTargets.Class |
+                    AttributeTargets.Struct |
+                    AttributeTargets.Enum |
+                    AttributeTargets.Constructor |
+                    AttributeTargets.Method |
+                    AttributeTargets.Property |
+                    AttributeTargets.Field |
+                    AttributeTargets.Event |
+                    AttributeTargets.Interface |
+                    AttributeTargets.Delegate, Inherited = false)]
```

Comment:
@terrajobst, @bartonjs, did we discuss Inherited false/true for this?  Seems like true might actually make more sense; if you derive from something that's experimental, seems you'd also need to be experimental?

Reply:
The rule we've been using for our version says that:

* You can mark a whole assembly as experimental and that affects every symbol exposed by the assembly.
* You can mark a base class as experimental and that affects every derived class.
* You can mark an outer type as experimental and that affects nested types.
* You can mark a virtual method as experimental and that affects every implementation of the method.

We decided not to do this for interface types however. If your type implements an experimental interface, it does not make your new type experimental. Although I think a point can be made that experimental annotations on interface types also should contaminate implementations.